src/main/decompInitMod.F90

```diff
@@ -571,291 +663,195 @@ subroutine decompInit_glcp(lni,lnj,glc_behavior)
        coCount(gi) = icohorts  ! number of fates cohorts for local gricell index gi
     enddo
 
-    ! Determine gstart, lstart, cstart, pstart, coStart for the OUTPUT 1d data structures
-
-    ! gather the gdc subgrid counts to masterproc in glo order
-    ! compute glo ordered start indices from the counts
-    ! scatter the subgrid start indices back out to the gdc gridcells
-    ! set the local gindex array for the subgrid from the subgrid start and count arrays
+    ! ---------------------------------------
+    ! Arrays needed to determine gindex_xxx(:)
+    ! ---------------------------------------
 
-    ng = mct_gsmap_gsize(gsmap_lnd_gdc2glo)
-    allocate(arrayglob(ng))
+    allocate(ioff(lsize_g))
 
-    arrayglob(:) = 0
-    call gather_data_to_master(gcount, arrayglob, grlnd)
     if (masterproc) then
-       val1 = arrayglob(1)
-       arrayglob(1) = 1
-       do n = 2,ng
-          val2 = arrayglob(n)
-          arrayglob(n) = arrayglob(n-1) + val1
-          val1 = val2
-       enddo
+       allocate (gridcells_per_pe(0:npes-1))
+    else
+       allocate(gridcells_per_pe(0))
     endif
-    call scatter_data_from_master(gstart, arrayglob, grlnd)
-
-    ! lstart for gridcell (n) is the total number of the landunits
-    ! over gridcells 1->n-1
+    call mpi_gather(lsize_g, 1, MPI_INTEGER, gridcells_per_pe, 1, MPI_INTEGER, 0, mpicom, ier)
```

Comment:
It does look like the spmd Gather-Scatter had only been used for decomposition initialization. But, we should think about this a bit, because there is an outstanding need for having additional communication in CTSM at irregular intervals (for seeds and other things that would be done at low intervals).

Reply:
The issue we have regarding this is in FATES: https://github.com/NGEET/fates/issues/471

Reply:
We did decide we do want this. We prefer moving to shr_mpi usage, but there will need to be more work to get this to happen.

Reply:
#1421